src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1575,6 +1575,118 @@
     "     check ((expire is not null and updated is not null)"
     "             or (expire is null and updated is null)))"]))
 
+(defn change-reports-hash-to-summary-hash
+  "Changes the reports table primary key hash to remove the events from
+  the hash calculation. This allows events to be garbage collected separately
+  from reports"
+  []
+  (jdbc/do-commands
+   ["CREATE TABLE reports_transform ("
+    "    id bigint not null,"
+    "    hash bytea not null,"
+    "    transaction_uuid uuid,"
+    "    certname text not null,"
+    "    puppet_version text not null,"
+    "    report_format smallint not null,"
+    "    configuration_version text not null,"
+    "    start_time timestamp with time zone not null,"
+    "    end_time timestamp with time zone not null,"
+    "    receive_time timestamp with time zone not null,"
+    "    noop boolean,"
+    "    environment_id bigint,"
+    "    status_id bigint,"
+    "    metrics_json json,"
+    "    logs_json json,"
+    "    producer_timestamp timestamp with time zone not null,"
+    "    metrics jsonb,"
+    "    logs jsonb,"
+    "    resources jsonb,"
+    "    catalog_uuid uuid,"
+    "    cached_catalog_status text,"
+    "    code_id text,"
+    "    producer_id bigint,"
+    "    noop_pending boolean,"
+    "    corrective_change boolean,"
+    "    job_id text)"])
+
+  (jdbc/call-with-query-rows
+   ["select
+       id, transaction_uuid, certname, puppet_version, report_format, configuration_version,
+       start_time, end_time, receive_time, noop, environment_id, status_id, metrics_json,
+       logs_json, producer_timestamp, metrics, logs, resources, catalog_uuid, cached_catalog_status,
+       code_id, producer_id, noop_pending, corrective_change, job_id
+     from reports"]
+   (fn [rows]
+     (reduce (fn [hashes-seen row]
+               (conj hashes-seen
+                     (let [hash (-> (hash/report-identity-hash row)
+                                    (sutils/munge-hash-for-storage))]
+                       (when-not (contains? hashes-seen hash)
```

Comment:
If you feel strongly about it, no worries, but I believe it's generally idiomatic to avoid contains? in favor of using the set itself as the function (e.g. `(hashes-seen hash)`.  Though I could see a clarity argument on the other side if the set name's not sufficiently evocative...

Reply:
changed to `(hashes-seen hash-str)`